src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/RazorRenameProvider.ts

```diff
@@ -0,0 +1,66 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as vscode from 'vscode';
+import { RazorDocumentManager } from './RazorDocumentManager';
+import { RazorDocumentSynchronizer } from './RazorDocumentSynchronizer';
+import { RazorLanguageFeatureBase } from './RazorLanguageFeatureBase';
+import { RazorLanguageServiceClient } from './RazorLanguageServiceClient';
+import { RazorLogger } from './RazorLogger';
+import { LanguageKind } from './RPC/LanguageKind';
+
+export class RazorRenameProvider
+    extends RazorLanguageFeatureBase
+    implements vscode.RenameProvider {
+
+    constructor(
+        documentSynchronizer: RazorDocumentSynchronizer,
+        documentManager: RazorDocumentManager,
+        serviceClient: RazorLanguageServiceClient,
+        logger: RazorLogger) {
+        super(documentSynchronizer, documentManager, serviceClient, logger);
+    }
+
+    public async prepareRename(
+        document: vscode.TextDocument,
+        position: vscode.Position,
+        token: vscode.CancellationToken) {
+
+        const projection = await this.getProjection(document, position, token);
+        if (!projection || projection.languageKind !== LanguageKind.CSharp) {
+            // We only support C# renames for now.
+            return Promise.reject('Cannot rename this symbol.');
+        }
+
+        return;
```

Comment:
Yes, returning nothing here will actually let the rename go through which is what we want here. If this behavior changes, we can update this.

Reply:
Can just not return. Does same thing 😄